dataquest-dev/DSpace
# feature/pid-3-return-directly-cmdi

| Phases            | MM  |  MB  | MR |   JM | Total  |
|-----------------|----:|----:|-----:|-----:|-------:|
| ETA                  |  0 |   6 |     0 |      0 |        0 |
| Developing      |  0  |   20 |    0 |      0 |         0 |
| Review             |  0  |    0 |    0 |      0 |         0 |
| Total                |   -  |   -   |  -    |   -    |         0 |
| ETA est.             |      |       |       |         |         0 |
| ETA cust.           |   -  |   -  |   -   |   -     |        0 |

## How to test it

Similar to https://github.com/dataquest-dev/DSpace/pull/120 , just check cmdi format [here](https://github.com/dataquest-dev/DSpace/blob/feature/pid-3-return-directly-cmdi/dspace/config/crosswalks/oai/metadataFormats/lindat_cmdi.xsl)

In special case, uploaded file must be returned:
- [ ] file to be returned [the_good_cmdi.txt](https://github.com/dataquest-dev/DSpace/files/9368239/the_good_cmdi.txt)
- [ ] change extension of file so that it's called `something.xml`
- [ ] create item on which you are going to test it
- [ ] add metadata local.hasCMDI = true
- [ ] add bundle METADATA and upload that file

![image](https://user-images.githubusercontent.com/88670521/185314367-6c8c54da-dd32-4036-8a06-81e0143afbe8.png)

![image](https://user-images.githubusercontent.com/88670521/185314541-b9bb650b-60ba-47d9-bb06-aa5962b73e51.png)

![image](https://user-images.githubusercontent.com/88670521/185315041-830124bb-c107-43bd-8fa2-96680f79fa75.png)

- [ ] don't forget to call `dspace oai import -c` 
- [ ] when opening OAIPMH on this file, metadata must contain contents of that file exactly

## Problem description
Add cmdi format
### Reported issues
copy xsl for cmdi format from clarin-dspace repository
### Not-reported issues


new dspace can't use java
java is required for this format
when forced use of java, it cannot be autowired
other unknown xalan functions in xsl
multiple files neede to be imported
java classes use external files which were problematic to load

## Problems
most are listed in non-reported issues
